components/shortcuts/shortcuts_modal/shortcuts_modal.tsx

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {injectIntl, defineMessages, IntlShape} from 'react-intl';
+
+import ModalStore from 'stores/modal_store';
+import Constants from 'utils/constants';
+
+import * as Utils from 'utils/utils';
+
+import {t} from 'utils/i18n';
+
+import {shortcuts} from '../shortcuts';
+
+import ShortcutSequence from '../shortcut_sequence';
+
+import './shortcuts_modal.scss';
+
+const modalMessages = defineMessages({
+    msgHeader: {
+        id: t('shortcuts.msgs.header'),
+        defaultMessage: 'Messages',
+    },
+    msgInputHeader: {
+        id: t('shortcuts.msgs.input.header'),
+        defaultMessage: 'Works inside an empty input field',
+    },
+    filesHeader: {
+        id: t('shortcuts.files.header'),
+        defaultMessage: 'Files',
+    },
+    browserHeader: {
+        id: t('shortcuts.browser.header'),
+        defaultMessage: 'Built-in Browser Commands',
+    },
+    info: {
+        id: t('shortcuts.info'),
+        defaultMessage:
+      'Begin a message with / for a list of all the commands at your disposal.',
+    },
+    navHeader: {
+        id: t('shortcuts.nav.header'),
+        defaultMessage: 'Navigation',
+    },
+});
+
+type Props = {
+    intl: IntlShape;
+}
+
+type State = {
+    show: boolean;
+}
+
+class ShortcutsModal extends React.PureComponent<Props, State> {
```

Comment:
why is this built using a class-based component? Especially since this is a new component!